sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/InsertIntoHiveTable.scala

```diff
@@ -240,54 +240,49 @@ case class InsertIntoHiveTable(
     // holdDDLTime will be true when TOK_HOLD_DDLTIME presents in the query as a hint.
     val holdDDLTime = false
     if (partition.nonEmpty) {
-
-      // loadPartition call orders directories created on the iteration order of the this map
-      val orderedPartitionSpec = new util.LinkedHashMap[String, String]()
-      table.hiveQlTable.getPartCols.asScala.foreach { entry =>
-        orderedPartitionSpec.put(entry.getName, partitionSpec.getOrElse(entry.getName, ""))
-      }
-
-      // inheritTableSpecs is set to true. It should be set to false for an IMPORT query
-      // which is currently considered as a Hive native command.
-      val inheritTableSpecs = true
-      // TODO: Correctly set isSkewedStoreAsSubdir.
-      val isSkewedStoreAsSubdir = false
       if (numDynamicPartitions > 0) {
-        client.synchronized {
-          client.loadDynamicPartitions(
-            outputPath.toString,
-            table.catalogTable.qualifiedName,
-            orderedPartitionSpec,
-            overwrite,
-            numDynamicPartitions,
-            holdDDLTime,
-            isSkewedStoreAsSubdir)
-        }
+        externalCatalog.loadDynamicPartitions(
+          db = table.catalogTable.database,
+          table = table.catalogTable.identifier.table,
+          outputPath.toString,
+          partitionSpec,
+          overwrite,
+          numDynamicPartitions,
+          holdDDLTime = holdDDLTime,
+          listBucketingEnabled = false)
```

Comment:
uh... `isSkewedStoreAsSubdir` and `listBucketingEnabled` are the same!!! `loadDynamicPartitions` is using [`listBucketingEnabled`](https://github.com/apache/hive/blob/ecab0d072b50a8d85dca6e850e47425d96c1ac09/ql/src/java/org/apache/hadoop/hive/ql/metadata/Hive.java#L1773), but `loadPartition` is using [`isSkewedStoreAsSubdir`](https://github.com/apache/hive/blob/ecab0d072b50a8d85dca6e850e47425d96c1ac09/ql/src/java/org/apache/hadoop/hive/ql/metadata/Hive.java#L1474). 

`loadDynamicPartitions` eventually passes the value of `listBucketingEnabled` to `isSkewedStoreAsSubdir` when [calling `loadPartition`](https://github.com/apache/hive/blob/ecab0d072b50a8d85dca6e850e47425d96c1ac09/ql/src/java/org/apache/hadoop/hive/ql/metadata/Hive.java#L1817). 

To differ list bucketing tables from normal skewed tables, they use an optional parameter `store as DIRECTORIES`. See the JIRA [HIVE-3649](https://issues.apache.org/jira/browse/HIVE-3649). 

Actually, we can remove TODO for either `isSkewedStoreAsSubdir` or `loadDynamicPartitions`, because we always [abend this feature](https://github.com/apache/spark/blob/1f06a5b6a0584d0c9656f58eaf54e54e2383c82b/sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala#L367-L369) in `getTableOption`